libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -9,26 +9,46 @@ import {
 } from '@tmo/books/data-access';
 import { FormBuilder } from '@angular/forms';
 import { Book } from '@tmo/shared/models';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'tmo-book-search',
   templateUrl: './book-search.component.html',
   styleUrls: ['./book-search.component.scss']
 })
-export class BookSearchComponent {
+export class BookSearchComponent implements OnInit, OnDestroy {
   books: ReadingListBook[];
 
   books$ = this.store.select(getAllBooks);
 
   searchForm = this.fb.group({
     term: ''
   });
+  unsubscribe$: Subject<void> = new Subject<void>();
 
   constructor(
     private readonly store: Store,
     private readonly fb: FormBuilder
   ) {}
 
+  ngOnInit(): void {
+    // search for books as user typing
+    this.searchForm.controls['term'].valueChanges.pipe(
+      debounceTime(500),
+      distinctUntilChanged(),
+      takeUntil(this.unsubscribe$))
+      .subscribe(() => this.searchBooks(),
+        err => console.error(err, 'Error in searching books')
+      );
+  }
+
+  ngOnDestroy(): any {
+    // unsubscribe from observable
```

Comment:
Please don't use comments like this. Comments should say _why_ you're doing a thing. The code tells you _what_ you're doing.

Reply:
Updated. Thank you!